docs/fundamentals/as-avs.txt

```diff
@@ -0,0 +1,262 @@
+
+You must include the following import statement in your application to
+create a ``SearchOperatorInterface`` instance:
+
+.. code-block:: php
+
+   use MongoDB\Builder\Search;
+
+The following code uses the index created in the preceding
+:ref:`laravel-as-index` section to perform an Atlas Search query on the
+``movies`` collection:
+
+.. io-code-block::
+   :copyable: true
+
+   .. input:: /includes/fundamentals/as-avs/AtlasSearchTest.php
+      :language: php
+      :dedent:
+      :start-after: start-search-query
+      :end-before: end-search-query
+
+   .. output::
+      :language: json
+      :visible: false
+
+      [
+         { "title": "Dreaming of Jakarta",
+           "year": 1990
+         },
+         { "title": "See You in My Dreams",
+           "year": 1996
+         }
+      ]
+
+You can use the ``search()`` method to perform many types of Atlas
+Search queries. Depending on your desired query, you can pass the
+following optional parameters to ``search()``:
+
+- ``index``: ``string``
+- ``highlight``: ``array``
+- ``concurrent``: ``bool``
+- ``count``: ``string``
+- ``searchAfter``: ``string``
+- ``searchBefore`` ``string``
+- ``scoreDetails``: ``bool``
+- ``sort``: ``array``
+- ``returnStoredSource``: ``bool``
+- ``tracking``: ``array``
+
+To learn more about these parameters, see the :atlas:`Fields
+</atlas-search/aggregation-stages/search/#fields>` section of the
+``$search`` operator reference in the Atlas documentation.
+
+Autocomplete Queries
+````````````````````
+
+The {+odm-short+} provides the ``autocomplete()`` method as a query
+builder method and as an Eloquent model method. You can use the
+``autocomplete()`` method to run autocomplete searches on documents in your
+collections. This method returns only the values of the field you
+specify as the query path.
+
+To learn more about this type of Atlas Search query, see the
+:atlas:`autocomplete </atlas-search/autocomplete/>` reference in the
+Atlas documentation.
+
+.. note::
+
+   You must create an Atlas Search index with an autocomplete configuration
+   on your collection before you can perform autocomplete searches. See the
+   :ref:`laravel-as-index` section of this guide to learn more about
+   creating Search indexes.
+
+The following code performs an Atlas Search autocomplete query for the
+string ``"jak"`` on the ``title`` field:
+
+.. io-code-block::
+   :copyable: true
+
+   .. input:: /includes/fundamentals/as-avs/AtlasSearchTest.php
+      :language: php
+      :dedent:
+      :start-after: start-auto-query
+      :end-before: end-auto-query
+
+   .. output::
+      :language: json
+      :visible: false
+
+      [
+         "Dreaming of Jakarta",
+         "Jakob the Liar",
+         "Emily Calling Jake"
+      ]
+
+You can also pass the following parameters to the ``autocomplete()``
+method to customize the query:
+
+- ``fuzzy``: ``bool`` or ``array`` (default: ``false``)
+- ``tokenOrder``: ``string`` (default: ``'any'``)
```

Comment:
S: Same here! I would create a table with descriptions of both parameters.

Reply:
See earlier reply